From 024f99022be0f9ebcb22d29c8fd3f3c4243ff519 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Tue, 25 Jun 2024 12:26:30 -0700
Subject: [PATCH] Docs/translations/ko_KR: Link memory-barriers wrapper with
 other language versions

The menu for documents of other available languages comes up only if the
documents are in same file hierarchy.  Because memory-barriers.txt of
Korean translation is at the root index while that for English is under
core-api/wrappers/memory-barriers.rst, the Korean translation is not
linked with other available language version via the menu.  Move the
document under the same directory hierarchy to make it linked with other
language versions.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 .../core-api/wrappers/memory-barriers.rst      | 18 ++++++++++++++++++
 Documentation/translations/ko_KR/index.rst     | 14 ++------------
 2 files changed, 20 insertions(+), 12 deletions(-)
 create mode 100644 Documentation/translations/ko_KR/core-api/wrappers/memory-barriers.rst

diff --git a/Documentation/translations/ko_KR/core-api/wrappers/memory-barriers.rst b/Documentation/translations/ko_KR/core-api/wrappers/memory-barriers.rst
new file mode 100644
index 000000000000..526ae534dd86
--- /dev/null
+++ b/Documentation/translations/ko_KR/core-api/wrappers/memory-barriers.rst
@@ -0,0 +1,18 @@
+.. SPDX-License-Identifier: GPL-2.0
+   This is a simple wrapper to bring memory-barriers.txt into the RST world
+   until such a time as that file can be converted directly.
+
+=========================
+리눅스 커널 메모리 배리어
+=========================
+
+.. raw:: latex
+
+    \footnotesize
+
+.. include:: ../../memory-barriers.txt
+    :literal:
+
+.. raw:: latex
+
+    \normalsize
diff --git a/Documentation/translations/ko_KR/index.rst b/Documentation/translations/ko_KR/index.rst
index 2d51f1481310..a20772f9d61c 100644
--- a/Documentation/translations/ko_KR/index.rst
+++ b/Documentation/translations/ko_KR/index.rst
@@ -12,18 +12,8 @@
    :maxdepth: 1
 
    process/howto
-
-
-리눅스 커널 메모리 배리어
--------------------------
-
-.. raw:: latex
-
-    \footnotesize
-
-.. include:: ./memory-barriers.txt
-    :literal:
+   core-api/wrappers/memory-barriers.rst
 
 .. raw:: latex
 
-    }\kerneldocEndKR
+       }\kerneldocEndKR
-- 
2.39.2

